src/components/AccordionSection/index.js

```diff
@@ -0,0 +1,176 @@
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import RenderIf from './../RenderIf';
+import { uniqueId } from './../../libs/utils';
+import { Consumer } from './../Accordion/context';
+import ButtonIcon from './../ButtonIcon';
+import RightArrow from './rightArrow';
+import isInArray from './check-is-in-array';
+import './styles.css';
+
+class AccordionItem extends Component {
+    constructor(props) {
+        super(props);
+        this.accordionDetailsId = uniqueId('accordion-details');
+        this.name = uniqueId('accordion-section');
+        this.handleSelect = this.handleSelect.bind(this);
+    }
+
+    getContainerClassNames() {
+        const { className, disabled } = this.props;
+        return classnames('rainbow-accordion_container', {
+            'rainbow-accordion_container--disabled': disabled,
+        }, className);
+    }
+
+    getTabIndex() {
+        const { disabled } = this.props;
+        if (disabled) {
+            return -1;
+        }
+        return undefined;
+    }
+
+    resolveNamesWhenMultiple() {
+        const { name, activeNames } = this.props;
+        const nameToToggle = name || this.name;
+
+        if (activeNames === undefined) {
+            return [nameToToggle];
+        }
+        if (isInArray(activeNames, nameToToggle)) {
+            return activeNames;
+        }
+        return activeNames.concat([nameToToggle]);
+    }
+
+    resolveNames() {
+        const { name, multiple } = this.props;
+        const nameToToggle = name || this.name;
+        if (multiple) {
+            return this.resolveNamesWhenMultiple();
+        }
+        return nameToToggle;
+    }
+
+    handleSelect(event) {
+        const { disabled, privateOnToggleSection } = this.props;
+        if (!disabled) {
+            privateOnToggleSection(event, this.resolveNames());
+        }
+    }
+
+    isExpanded() {
+        const { activeNames, name, multiple } = this.props;
+        const currentName = name || this.name;
+        if (multiple && Array.isArray(activeNames)) {
+            return isInArray(activeNames, currentName);
+        }
+        return activeNames === currentName;
+    }
+
+    render() {
+        const {
+            style,
+            disabled,
+            children,
+            label,
+            icon,
+            assistiveText,
+        } = this.props;
+
+        return (
+            <li className={this.getContainerClassNames()} style={style} disabled={disabled}>
+                <section>
+                    <div className="rainbow-accordion_summary">
+                        <h3 className="rainbow-accordion_summary-heading">
+                            <button
+                                className="rainbow-accordion_summary-button-heading"
+                                tabIndex={this.getTabIndex()}
+                                aria-controls={this.accordionDetailsId}
+                                aria-expanded={this.isSelected}
```

Comment:
this.isSelected does't exist in this class component then it will be always undefined. Here you must to use this.isExpanded() instead.